plugins/org.eclipse.persistence.asm/pom.xml

```diff
@@ -110,6 +253,63 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-antrun-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>fix-asm</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                        <configuration>
+                            <!-- ASM itself uses 1.8 source level and then retrofits the bytecode
+                            to work on Java SE 5+. We don't use retrofitter, so we need
+                            to fix sources to compile for Java SE 7 (EclipseLink 2.6.x) -->
+                            <target>
+                                <replace file="${project.build.directory}/generated-sources/asm-src/org/eclipse/persistence/internal/libraries/asm/tree/Util.java">
+                                    <replacefilter>
+                                        <replacetoken><![CDATA[new ArrayList<>(1)]]></replacetoken>
+                                        <replacevalue><![CDATA[new ArrayList<T>(1)]]></replacevalue>
+                                    </replacefilter>
+                                </replace>
```

Comment:
Interesting I believe the best solution from possible options, but I think, that should be applied to all *.java files in `generated-sources` to cover possible ASM code changes.

Reply:
that would have 2 drawbacks:
* it's faster to change 1 file than XX files
* it's probably better to "track" necessary changes, git won't do it for us here